devito/operator/operator.py

```diff
@@ -128,100 +127,102 @@ class Operator(Callable):
     _default_includes = ['stdlib.h', 'math.h', 'sys/time.h']
     _default_globals = []
 
-    def __init__(self, expressions, **kwargs):
+    def __new__(cls, expressions, **kwargs):
+        if expressions is None:
+            # Return a dummy Callable. This is exploited by unpickling. Users
+            # can't do anything useful with it
+            return super(Operator, cls).__new__(cls, **kwargs)
+
+        with timed_region('op-compile') as r:
+            op = cls._compile(expressions, **kwargs)
+        op._profiler.py_timers.update(r.timings)
+
+        op._emit_compile_profiling()
+
+        return op
+
+    @classmethod
+    def _compile(cls, expressions, **kwargs):
```

Comment:
Not sure an appropriate name, des more than compile, maybe `_build`?

Reply:
fair enough, changing